File: snapshots/output-net10.0/syntax/VBMain/Classes.vb
  Imports System.Diagnostics.CodeAnalysis
'         ^^^^^^ read_access scip-dotnet nuget . . System/
'                enclosing_range 0:0-0:39
'         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ import scip-dotnet nuget . . CodeAnalysis/
'                                         enclosing_range 0:0-0:39
'                ^^^^^^^^^^^ read_access scip-dotnet nuget . . Diagnostics/
'                            enclosing_range 0:0-0:39
'                            ^^^^^^^^^^^^ read_access scip-dotnet nuget . . CodeAnalysis/
'                                         enclosing_range 0:0-0:39

  Namespace VBMain
'           ^^^^^^ read_access scip-dotnet nuget . . VBMain/
'                  enclosing_range 2:0-2:16
      <SuppressMessage("ReSharper", "all")>
'      ^^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 CodeAnalysis/SuppressMessageAttribute#`.ctor`().
'                      enclosing_range 3:4-4:24
      Public Class Classes
'                  ^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#
'                          enclosing_range 3:4-77:13
'                          documentation ```vb\nClass Classes\n```
'                          kind Class

          Public Name As String
'                ^^^^ definition scip-dotnet nuget . . VBMain/Classes#Name.
'                     enclosing_range 6:8-6:29
'                     documentation ```vb\nPublic Classes.Name As String\n```
'                     kind Field
'                     relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#
          Public Const IntConstant As Integer = 1
'                      ^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#IntConstant.
'                                  enclosing_range 7:8-7:47
'                                  documentation ```vb\nPublic Const Classes.IntConstant As Integer = 1\n```
'                                  kind Constant
'                                  relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
          Public Const StringConstant As String = "hello"
'                      ^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#StringConstant.
'                                     enclosing_range 8:8-8:55
'                                     documentation ```vb\nPublic Const Classes.StringConstant As String = "hello"\n```
'                                     kind Constant
'                                     relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#

          Public Sub New(ByVal name As Integer)
'                    ^^^ definition scip-dotnet nuget . . VBMain/Classes#`.ctor`().
'                        enclosing_range 10:8-12:15
'                        documentation ```vb\nPublic Sub Classes.New(name As Integer)\n```
'                        kind Constructor
'                              ^^^^ definition scip-dotnet nuget . . VBMain/Classes#`.ctor`().(name)
'                                   enclosing_range 10:8-10:45
'                                   documentation ```vb\nname As Integer\n```
'                                   kind Parameter
'                                   relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
              Me.Name = "name"
'                ^^^^ read_access scip-dotnet nuget . . VBMain/Classes#Name.
'                     enclosing_range 10:8-12:15
          End Sub

          Public Sub New(ByVal name As String)
'                    ^^^ definition scip-dotnet nuget . . VBMain/Classes#`.ctor`(+1).
'                        enclosing_range 14:8-16:15
'                        documentation ```vb\nPublic Sub Classes.New(name As String)\n```
'                        kind Constructor
'                              ^^^^ definition scip-dotnet nuget . . VBMain/Classes#`.ctor`(+1).(name)
'                                   enclosing_range 14:8-14:44
'                                   documentation ```vb\nname As String\n```
'                                   kind Parameter
'                                   relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#
              Me.Name = name
'                ^^^^ read_access scip-dotnet nuget . . VBMain/Classes#Name.
'                     enclosing_range 14:8-16:15
'                       ^^^^ read_access scip-dotnet nuget . . VBMain/Classes#`.ctor`(+1).(name)
'                            enclosing_range 14:8-16:15
          End Sub

          Protected Overrides Sub Finalize()
'                                 ^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#Finalize().
'                                          enclosing_range 18:8-20:15
'                                          documentation ```vb\nProtected Overrides Sub Classes.Finalize()\n```
'                                          kind Method
'                                          relationship implementation reference scip-dotnet nuget System.Runtime 10.0.0.0 System/Object#Finalize().
              Console.WriteLine(42)
'             ^^^^^^^ read_access scip-dotnet nuget System.Console 10.0.0.0 System/Console#
'                     enclosing_range 18:8-20:15
'                     ^^^^^^^^^ read_access scip-dotnet nuget System.Console 10.0.0.0 System/Console#WriteLine(+7).
'                               enclosing_range 18:8-20:15
          End Sub

          Public Class ObjectClass
'                      ^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#ObjectClass#
'                                  enclosing_range 22:8-25:17
'                                  documentation ```vb\nClass ObjectClass\n```
'                                  kind Class
'                                  relationship implementation scip-dotnet nuget . . VBMain/Classes#SomeInterface#
              Inherits Object
              Implements SomeInterface
'                        ^^^^^^^^^^^^^ read_access scip-dotnet nuget . . VBMain/Classes#SomeInterface#
'                                      enclosing_range 24:12-24:36
          End Class

          Public Partial Class PartialClass
'                              ^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#PartialClass#
'                                           enclosing_range 27:8-28:17
'                                           documentation ```vb\nClass PartialClass\n```
'                                           kind Class
          End Class

          Class TypeParameterClass(Of T)
'               ^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#TypeParameterClass#
'                                  enclosing_range 30:8-31:17
'                                  documentation ```vb\nClass TypeParameterClass(Of T)\n```
'                                  kind Class
'                                     ^ definition local 0
'                                       enclosing_range 30:8-30:38
'                                       documentation ```vb\nT\n```
'                                       kind TypeParameter
          End Class

          Friend Class InternalMultipleTypeParametersClass(Of T1, T2)
'                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#InternalMultipleTypeParametersClass#
'                                                          enclosing_range 33:8-34:17
'                                                          documentation ```vb\nClass InternalMultipleTypeParametersClass(Of T1, T2)\n```
'                                                          kind Class
'                                                             ^^ definition local 1
'                                                                enclosing_range 33:8-33:67
'                                                                documentation ```vb\nT1\n```
'                                                                kind TypeParameter
'                                                                 ^^ definition local 2
'                                                                    enclosing_range 33:8-33:67
'                                                                    documentation ```vb\nT2\n```
'                                                                    kind TypeParameter
          End Class

          Interface ICovariantContravariant(Of In T1, Out T2)
'                   ^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#ICovariantContravariant#
'                                           enclosing_range 36:8-41:21
'                                           documentation ```vb\nInterface ICovariantContravariant(Of In T1, Out T2)\n```
'                                           kind Interface
'                                                 ^^ definition local 3
'                                                    enclosing_range 36:8-36:59
'                                                    documentation ```vb\nIn T1\n```
'                                                    kind TypeParameter
'                                                         ^^ definition local 4
'                                                            enclosing_range 36:8-36:59
'                                                            documentation ```vb\nOut T2\n```
'                                                            kind TypeParameter
              Sub Method1(ByVal t1 As T1)
'                 ^^^^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Classes#ICovariantContravariant#Method1().
'                         enclosing_range 36:8-41:21
'                         documentation ```vb\nSub ICovariantContravariant(Of In T1, Out T2).Method1(t1 As T1)\n```
'                         kind AbstractMethod
'                               ^^ definition scip-dotnet nuget . . VBMain/Classes#ICovariantContravariant#Method1().(t1)
'                                  enclosing_range 37:12-37:39
'                                  documentation ```vb\nt1 As T1\n```
'                                  kind Parameter
'                                  relationship type_definition local 3
'                                     ^^ read_access local 3
'                                        enclosing_range 37:12-37:39

              Function Method2() As T2
'                      ^^^^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Classes#ICovariantContravariant#Method2().
'                              enclosing_range 36:8-41:21
'                              documentation ```vb\nFunction ICovariantContravariant(Of In T1, Out T2).Method2() As T2\n```
'                              kind AbstractMethod
'                                   ^^ read_access local 4
'                                      enclosing_range 39:12-39:36

          End Interface

          Public Class StructConstraintClass(Of T As Structure)
'                      ^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#StructConstraintClass#
'                                            enclosing_range 43:8-44:17
'                                            documentation ```vb\nClass StructConstraintClass(Of T As Structure)\n```
'                                            kind Class
'                                               ^ definition local 5
'                                                 enclosing_range 43:8-43:61
'                                                 documentation ```vb\nT\n```
'                                                 kind TypeParameter
          End Class

          Public Class ClassConstraintClass(Of T As Class)
'                      ^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#ClassConstraintClass#
'                                           enclosing_range 46:8-47:17
'                                           documentation ```vb\nClass ClassConstraintClass(Of T As Class)\n```
'                                           kind Class
'                                              ^ definition local 6
'                                                enclosing_range 46:8-46:56
'                                                documentation ```vb\nT\n```
'                                                kind TypeParameter
          End Class

          Public Class NewConstraintClass(Of T As New)
'                      ^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#NewConstraintClass#
'                                         enclosing_range 49:8-50:17
'                                         documentation ```vb\nClass NewConstraintClass(Of T As New)\n```
'                                         kind Class
'                                            ^ definition local 7
'                                              enclosing_range 49:8-49:52
'                                              documentation ```vb\nT\n```
'                                              kind TypeParameter
          End Class

          Public Class TypeParameterConstraintClass(Of T As SomeInterface)
'                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#TypeParameterConstraintClass#
'                                                   enclosing_range 52:8-53:17
'                                                   documentation ```vb\nClass TypeParameterConstraintClass(Of T As SomeInterface)\n```
'                                                   kind Class
'                                                      ^ definition local 8
'                                                        enclosing_range 52:8-52:72
'                                                        documentation ```vb\nT\n```
'                                                        kind TypeParameter
'                                                           ^^^^^^^^^^^^^ read_access scip-dotnet nuget . . VBMain/Classes#SomeInterface#
'                                                                         enclosing_range 52:8-52:72
          End Class

          Private Class MultipleTypeParameterConstraintsClass(Of T1 As {SomeInterface, SomeInterface2, New}, T2 As SomeInterface2)
'                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#MultipleTypeParameterConstraintsClass#
'                                                             enclosing_range 55:8-56:17
'                                                             documentation ```vb\nClass MultipleTypeParameterConstraintsClass(Of T1 As {SomeInterface, SomeInterface2, New}, T2 As SomeInterface2)\n```
'                                                             kind Class
'                                                                ^^ definition local 9
'                                                                   enclosing_range 55:8-55:128
'                                                                   documentation ```vb\nT1\n```
'                                                                   kind TypeParameter
'                                                                       ^^^^^^^^^^^^^ read_access scip-dotnet nuget . . VBMain/Classes#SomeInterface#
'                                                                                     enclosing_range 55:8-55:128
'                                                                                      ^^^^^^^^^^^^^^ read_access scip-dotnet nuget . . VBMain/Classes#SomeInterface2#
'                                                                                                     enclosing_range 55:8-55:128
'                                                                                                            ^^ definition local 10
'                                                                                                               enclosing_range 55:8-55:128
'                                                                                                               documentation ```vb\nT2\n```
'                                                                                                               kind TypeParameter
'                                                                                                                  ^^^^^^^^^^^^^^ read_access scip-dotnet nuget . . VBMain/Classes#SomeInterface2#
'                                                                                                                                 enclosing_range 55:8-55:128
          End Class

          Class IndexClass
'               ^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#IndexClass#
'                          enclosing_range 58:8-69:17
'                          documentation ```vb\nClass IndexClass\n```
'                          kind Class
              Private a As Boolean
'                     ^ definition scip-dotnet nuget . . VBMain/Classes#IndexClass#a.
'                       enclosing_range 59:12-59:32
'                       documentation ```vb\nPrivate IndexClass.a As Boolean\n```
'                       kind Field
'                       relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Boolean#

              Default Public Property Item(ByVal index As Integer) As Boolean
'                                     ^^^^ definition scip-dotnet nuget . . VBMain/Classes#IndexClass#Item.
'                                          enclosing_range 61:12-68:24
'                                          documentation ```vb\nPublic Default Property IndexClass.Item(index As Integer) As Boolean\n```
'                                          kind Property
'                                          relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Boolean#
'                                                ^^^^^ definition scip-dotnet nuget . . VBMain/Classes#IndexClass#Item.(index)
'                                                      enclosing_range 61:12-61:75
'                                                      documentation ```vb\nindex As Integer\n```
'                                                      kind Parameter
'                                                      relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
                  Get
                      Return a
'                            ^ read_access scip-dotnet nuget . . VBMain/Classes#IndexClass#a.
'                              enclosing_range 62:16-64:23
                  End Get
                  Set(ByVal value As Boolean)
'                           ^^^^^ definition scip-dotnet nuget . . VBMain/Classes#IndexClass#set_Item().(value)
'                                 enclosing_range 65:16-65:43
'                                 documentation ```vb\nvalue As Boolean\n```
'                                 kind Parameter
'                                 relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Boolean#
                      a = value
'                     ^ write_access scip-dotnet nuget . . VBMain/Classes#IndexClass#a.
'                       enclosing_range 65:16-67:23
'                         ^^^^^ read_access scip-dotnet nuget . . VBMain/Classes#IndexClass#set_Item().(value)
'                               enclosing_range 65:16-67:23
                  End Set
              End Property
          End Class

          Interface SomeInterface
'                   ^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#SomeInterface#
'                                 enclosing_range 71:8-72:21
'                                 documentation ```vb\nInterface SomeInterface\n```
'                                 kind Interface
          End Interface

          Friend Interface SomeInterface2
'                          ^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#SomeInterface2#
'                                         enclosing_range 74:8-75:21
'                                         documentation ```vb\nInterface SomeInterface2\n```
'                                         kind Interface
          End Interface

      End Class

  End Namespace

File: snapshots/output-net10.0/syntax/Main/Classes.cs
  using System.Diagnostics.CodeAnalysis;
//      ^^^^^^ read_access scip-dotnet nuget . . System/
//             enclosing_range 0:6-0:24
//      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ import scip-dotnet nuget . . CodeAnalysis/
//                                      enclosing_range 0:6-0:37
//             ^^^^^^^^^^^ read_access scip-dotnet nuget . . Diagnostics/
//                         enclosing_range 0:6-0:24
//                         ^^^^^^^^^^^^ read_access scip-dotnet nuget . . CodeAnalysis/
//                                      enclosing_range 0:6-0:37

  namespace Main;
//          ^^^^ read_access scip-dotnet nuget . . Main/
//               enclosing_range 2:10-2:14

  [SuppressMessage("ReSharper", "all")]
// ^^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 CodeAnalysis/SuppressMessageAttribute#`.ctor`().
//                 enclosing_range 4:1-4:16
  public class Classes
//             ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#
//                     enclosing_range 4:0-92:1
//                     documentation ```cs\nclass Classes\n```
//                     kind Class
  {
      public string Name;
//                  ^^^^ definition scip-dotnet nuget . . Main/Classes#Name.
//                       enclosing_range 7:4-7:23
//                       documentation ```cs\npublic string Classes.Name\n```
//                       kind Field
//                       relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#
      public const int IntConstant = 1;
//                     ^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#IntConstant.
//                                 enclosing_range 8:4-8:37
//                                 documentation ```cs\npublic const int Classes.IntConstant = 1\n```
//                                 kind Constant
//                                 relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
      public const string StringConstant = $"hello";
//                        ^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#StringConstant.
//                                       enclosing_range 9:4-9:50
//                                       documentation ```cs\npublic const string Classes.StringConstant = "hello"\n```
//                                       kind Constant
//                                       relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#

      public Classes(int name)
//           ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#`.ctor`().
//                   enclosing_range 11:4-14:5
//                   documentation ```cs\npublic Classes.Classes(int name)\n```
//                   kind Constructor
//                       ^^^^ definition scip-dotnet nuget . . Main/Classes#`.ctor`().(name)
//                            enclosing_range 11:4-14:5
//                            documentation ```cs\nint name\n```
//                            kind Parameter
//                            relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
      {
          Name = "name";
//        ^^^^ write_access scip-dotnet nuget . . Main/Classes#Name.
//             enclosing_range 13:8-13:21
      }

      public Classes(string name) => Name = name;
//           ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#`.ctor`(+1).
//                   enclosing_range 16:4-16:47
//                   documentation ```cs\npublic Classes.Classes(string name)\n```
//                   kind Constructor
//                          ^^^^ definition scip-dotnet nuget . . Main/Classes#`.ctor`(+1).(name)
//                               enclosing_range 16:4-16:47
//                               documentation ```cs\nstring name\n```
//                               kind Parameter
//                               relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#
//                                   ^^^^ write_access scip-dotnet nuget . . Main/Classes#Name.
//                                        enclosing_range 16:35-16:46
//                                          ^^^^ read_access scip-dotnet nuget . . Main/Classes#`.ctor`(+1).(name)
//                                               enclosing_range 16:35-16:46

      ~Classes()
//     ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#Finalize().
//             enclosing_range 18:4-21:5
//             documentation ```cs\nprotected Classes.~Classes()\n```
//             kind Method
      {
          Console.WriteLine(42);
//        ^^^^^^^ read_access scip-dotnet nuget System.Console 10.0.0.0 System/Console#
//                enclosing_range 20:8-20:29
//                ^^^^^^^^^ read_access scip-dotnet nuget System.Console 10.0.0.0 System/Console#WriteLine(+7).
//                          enclosing_range 20:8-20:29
      }

      public class ObjectClass : object, SomeInterface
//                 ^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#ObjectClass#
//                             enclosing_range 23:4-25:5
//                             documentation ```cs\nclass ObjectClass\n```
//                             kind Class
//                             relationship implementation scip-dotnet nuget . . Main/SomeInterface#
//                                       ^^^^^^^^^^^^^ read_access scip-dotnet nuget . . Main/SomeInterface#
//                                                     enclosing_range 23:39-23:52
      {
      }

      public partial class PartialClass
//                         ^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#PartialClass#
//                                      enclosing_range 27:4-29:5
//                                      documentation ```cs\nclass PartialClass\n```
//                                      kind Class
      {
      }

      class TypeParameterClass<T>
//          ^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#TypeParameterClass#
//                             enclosing_range 31:4-33:5
//                             documentation ```cs\nclass TypeParameterClass<T>\n```
//                             kind Class
//                             ^ definition local 0
//                               enclosing_range 31:4-33:5
//                               documentation ```cs\nT\n```
//                               kind TypeParameter
      {
      }

      internal class InternalMultipleTypeParametersClass<T1, T2>
//                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#InternalMultipleTypeParametersClass#
//                                                       enclosing_range 35:4-37:5
//                                                       documentation ```cs\nclass InternalMultipleTypeParametersClass<T1, T2>\n```
//                                                       kind Class
//                                                       ^^ definition local 1
//                                                          enclosing_range 35:4-37:5
//                                                          documentation ```cs\nT1\n```
//                                                          kind TypeParameter
//                                                           ^^ definition local 2
//                                                              enclosing_range 35:4-37:5
//                                                              documentation ```cs\nT2\n```
//                                                              kind TypeParameter
      {
      }

      interface ICovariantContravariant<in T1, out T2>
//              ^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#ICovariantContravariant#
//                                      enclosing_range 39:4-50:5
//                                      documentation ```cs\ninterface ICovariantContravariant<in T1, out T2>\n```
//                                      kind Interface
//                                         ^^ definition local 3
//                                            enclosing_range 39:4-50:5
//                                            documentation ```cs\nin T1\n```
//                                            kind TypeParameter
//                                                 ^^ definition local 4
//                                                    enclosing_range 39:4-50:5
//                                                    documentation ```cs\nout T2\n```
//                                                    kind TypeParameter
      {
          public void Method1(T1 t1)
//                    ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#ICovariantContravariant#Method1().
//                            enclosing_range 41:8-44:9
//                            documentation ```cs\nvoid ICovariantContravariant<in T1, out T2>.Method1(T1 t1)\n```
//                            kind Method
//                            ^^ read_access local 3
//                               enclosing_range 41:28-41:30
//                               ^^ definition scip-dotnet nuget . . Main/Classes#ICovariantContravariant#Method1().(t1)
//                                  enclosing_range 41:8-44:9
//                                  documentation ```cs\nT1 t1\n```
//                                  kind Parameter
//                                  relationship type_definition local 3
          {
              Console.WriteLine(t1);
//            ^^^^^^^ read_access scip-dotnet nuget System.Console 10.0.0.0 System/Console#
//                    enclosing_range 43:12-43:33
//                    ^^^^^^^^^ read_access scip-dotnet nuget System.Console 10.0.0.0 System/Console#WriteLine(+9).
//                              enclosing_range 43:12-43:33
//                              ^^ read_access scip-dotnet nuget . . Main/Classes#ICovariantContravariant#Method1().(t1)
//                                 enclosing_range 43:30-43:32
          }

          public T2? Method2()
//               ^^ read_access local 4
//                  enclosing_range 46:15-46:18
//                   ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#ICovariantContravariant#Method2().
//                           enclosing_range 46:8-49:9
//                           documentation ```cs\nT2? ICovariantContravariant<in T1, out T2>.Method2()\n```
//                           kind Method
          {
              return default(T2);
//                           ^^ read_access local 4
//                              enclosing_range 48:19-48:30
          }
      }

      public class StructConstraintClass<T> where T : struct
//                 ^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#StructConstraintClass#
//                                       enclosing_range 52:4-54:5
//                                       documentation ```cs\nclass StructConstraintClass<T> where T : struct\n```
//                                       kind Class
//                                       ^ definition local 5
//                                         enclosing_range 52:4-54:5
//                                         documentation ```cs\nT\n```
//                                         kind TypeParameter
//                                                ^ read_access local 5
//                                                  enclosing_range 52:48-52:49
      {
      }

      public class UnmanagedConstraintClass<T> where T : unmanaged
//                 ^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#UnmanagedConstraintClass#
//                                          enclosing_range 56:4-58:5
//                                          documentation ```cs\nclass UnmanagedConstraintClass<T> where T : unmanaged\n```
//                                          kind Class
//                                          ^ definition local 6
//                                            enclosing_range 56:4-58:5
//                                            documentation ```cs\nT\n```
//                                            kind TypeParameter
//                                                   ^ read_access local 6
//                                                     enclosing_range 56:51-56:52
      {
      }

      public class ClassConstraintClass<T> where T : class
//                 ^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#ClassConstraintClass#
//                                      enclosing_range 60:4-62:5
//                                      documentation ```cs\nclass ClassConstraintClass<T> where T : class\n```
//                                      kind Class
//                                      ^ definition local 7
//                                        enclosing_range 60:4-62:5
//                                        documentation ```cs\nT\n```
//                                        kind TypeParameter
//                                               ^ read_access local 7
//                                                 enclosing_range 60:47-60:48
      {
      }

      public class NonNullableConstraintClass<T> where T : notnull
//                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#NonNullableConstraintClass#
//                                            enclosing_range 64:4-66:5
//                                            documentation ```cs\nclass NonNullableConstraintClass<T> where T : notnull\n```
//                                            kind Class
//                                            ^ definition local 8
//                                              enclosing_range 64:4-66:5
//                                              documentation ```cs\nT\n```
//                                              kind TypeParameter
//                                                     ^ read_access local 8
//                                                       enclosing_range 64:53-64:54
      {
      }

      public class NewConstraintClass<T> where T : new()
//                 ^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#NewConstraintClass#
//                                    enclosing_range 68:4-70:5
//                                    documentation ```cs\nclass NewConstraintClass<T> where T : new()\n```
//                                    kind Class
//                                    ^ definition local 9
//                                      enclosing_range 68:4-70:5
//                                      documentation ```cs\nT\n```
//                                      kind TypeParameter
//                                             ^ read_access local 9
//                                               enclosing_range 68:45-68:46
      {
      }

      public class TypeParameterConstraintClass<T> where T : SomeInterface
//                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#TypeParameterConstraintClass#
//                                              enclosing_range 72:4-74:5
//                                              documentation ```cs\nclass TypeParameterConstraintClass<T> where T : SomeInterface\n```
//                                              kind Class
//                                              ^ definition local 10
//                                                enclosing_range 72:4-74:5
//                                                documentation ```cs\nT\n```
//                                                kind TypeParameter
//                                                       ^ read_access local 10
//                                                         enclosing_range 72:55-72:56
//                                                           ^^^^^^^^^^^^^ read_access scip-dotnet nuget . . Main/SomeInterface#
//                                                                         enclosing_range 72:59-72:72
      {
      }

      private class MultipleTypeParameterConstraintsClass<T1, T2> where T1 : SomeInterface, SomeInterface2, new()
//                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#MultipleTypeParameterConstraintsClass#
//                                                        enclosing_range 76:4-79:5
//                                                        documentation ```cs\nclass MultipleTypeParameterConstraintsClass<T1, T2> where T1 : SomeInterface, SomeInterface2, new() where T2 : SomeInterface2\n```
//                                                        kind Class
//                                                        ^^ definition local 11
//                                                           enclosing_range 76:4-79:5
//                                                           documentation ```cs\nT1\n```
//                                                           kind TypeParameter
//                                                            ^^ definition local 12
//                                                               enclosing_range 76:4-79:5
//                                                               documentation ```cs\nT2\n```
//                                                               kind TypeParameter
//                                                                      ^^ read_access local 11
//                                                                         enclosing_range 76:70-76:72
//                                                                           ^^^^^^^^^^^^^ read_access scip-dotnet nuget . . Main/SomeInterface#
//                                                                                         enclosing_range 76:75-76:88
//                                                                                          ^^^^^^^^^^^^^^ read_access scip-dotnet nuget . . Main/SomeInterface2#
//                                                                                                         enclosing_range 76:90-76:104
          where T2 : SomeInterface2
//              ^^ read_access local 12
//                 enclosing_range 77:14-77:16
//                   ^^^^^^^^^^^^^^ read_access scip-dotnet nuget . . Main/SomeInterface2#
//                                  enclosing_range 77:19-77:33
      {
      }

      class IndexClass
//          ^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#IndexClass#
//                     enclosing_range 81:4-90:5
//                     documentation ```cs\nclass IndexClass\n```
//                     kind Class
      {
          private bool a;
//                     ^ definition scip-dotnet nuget . . Main/Classes#IndexClass#a.
//                       enclosing_range 83:8-83:23
//                       documentation ```cs\nprivate bool IndexClass.a\n```
//                       kind Field
//                       relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Boolean#

          public bool this[int index]
//                             ^^^^^ definition scip-dotnet nuget . . Main/Classes#IndexClass#`this[]`.(index)
//                                   enclosing_range 85:8-89:9
//                                   documentation ```cs\nint index\n```
//                                   kind Parameter
//                                   relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
          {
              get { return a; }
//                         ^ read_access scip-dotnet nuget . . Main/Classes#IndexClass#a.
//                           enclosing_range 87:25-87:26
              set { a = value; }
//                  ^ write_access scip-dotnet nuget . . Main/Classes#IndexClass#a.
//                    enclosing_range 88:18-88:27
//                      ^^^^^ read_access scip-dotnet nuget . . Main/Classes#IndexClass#set_Item().(value)
//                            enclosing_range 88:18-88:27
          }
      }

  }

  public interface SomeInterface
//                 ^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/SomeInterface#
//                               enclosing_range 94:0-96:1
//                               documentation ```cs\ninterface SomeInterface\n```
//                               kind Interface
  {
  }

  internal interface SomeInterface2
//                   ^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/SomeInterface2#
//                                  enclosing_range 98:0-100:1
//                                  documentation ```cs\ninterface SomeInterface2\n```
//                                  kind Interface
  {
  }
